app/src/main/java/com/github/bgabriel998/softwaredevproject/ProfileActivity.java

```diff
@@ -166,79 +240,50 @@ public void onComplete(Task<AuthResult> task) {
-        snackbar.show();
-    }
-
     /* UI METHODS */
 
     /**
      * Sets what is visible on UI based on if the user is signed in or not.
      */
     public void setUI() {
         if (account.isSignedIn()) setLoggedUI();
-        else setNotLoggedUI();
+        else {
+            resetUI();
+            findViewById(R.id.signInButton).setVisibility(View.VISIBLE);
+        }
     }
 
     /**
      * Sets what is visible on UI if the user is logged
      */
     public void setLoggedUI(){
-        setVisibilitySignInUI(true);
-        hideChangeUsernameButtons();
+        resetUI();
+        findViewById(R.id.signOutButton).setVisibility(View.VISIBLE);
+        findViewById(R.id.changeUsernameButton).setVisibility(View.VISIBLE);
+        findViewById(R.id.addFriendButton).setVisibility(View.VISIBLE);
+        findViewById(R.id.friendsButton).setVisibility(View.VISIBLE);
     }
 
     /**
-     * Sets what is visible on UI if the user is not logged
+     * Hide all the buttons
      */
-    public void setNotLoggedUI(){
-        setVisibilitySignInUI(false);
-        hideChangeUsernameButtons();
-    }
-
-    /**
-     * Sets visibility for sign-in related buttons
-     * @param logged
-     */
-    public void setVisibilitySignInUI(Boolean logged) {
-        findViewById(R.id.signInButton).setVisibility(logged ? View.GONE : View.VISIBLE);
-        findViewById(R.id.signOutButton).setVisibility(logged ? View.VISIBLE : View.GONE);
-        findViewById(R.id.changeUsernameButton).setVisibility(logged ? View.VISIBLE : View.GONE);
-    }
-
-    /**
-     * Hides the username change buttons
-     */
-    public void hideChangeUsernameButtons() {
+    public void resetUI() {
+        findViewById(R.id.submitFriendButton).setVisibility(View.GONE);
+        findViewById(R.id.editTextFriend).setVisibility(View.GONE);
         findViewById(R.id.submitUsernameButton).setVisibility(View.GONE);
         findViewById(R.id.editTextUsername).setVisibility(View.GONE);
+        findViewById(R.id.signInButton).setVisibility(View.GONE);
+        findViewById(R.id.signOutButton).setVisibility(View.GONE);
+        findViewById(R.id.changeUsernameButton).setVisibility(View.GONE);
+        findViewById(R.id.addFriendButton).setVisibility(View.GONE);
+        findViewById(R.id.friendsButton).setVisibility(View.GONE);
     }
 
     /**
      * Sets what is visible on UI after a username change is requested or required
      */
     public void setUsernameChoiceUI() {
-        findViewById(R.id.signInButton).setVisibility(View.GONE);
-        findViewById(R.id.signOutButton).setVisibility(View.GONE);
-        findViewById(R.id.changeUsernameButton).setVisibility(View.GONE);
+        resetUI();
         findViewById(R.id.submitUsernameButton).setVisibility(View.VISIBLE);
         findViewById(R.id.editTextUsername).setVisibility(View.VISIBLE);
```

Comment:
Suggestion: add a boolean input parameter to decide if `submitUsernameButton` `editTextUsername` should be hidden or not. These two view are set to `View.GONE` and `View.VISIBLE` direclty after that

Reply:
Added booleans as input for resetUI.